changing the input arguments datatype (of summary stats) from data frame to dictionary

Ati-74 opened the pull request.

Ati-74 committed:
output of image processing repo script [sample:K-12]

K-12 images are processed by CP and the final csv . is used as an input for the image processing repo script and these are the output but in pickle format

Ati-74 committed:
Create .gitignore

Ati-74 committed:
unit test

when the script has unit test, a user just need to run the unit test (also follow its instruction)

Ati-74 committed:
Density Calc.

This code is written by J.C. & A.Y.

Ati-74 committed:
growth-rate-stat

This code is written by A.Y.

Ati-74 committed:
Anisotropy and Aspect ratio calc.

Inputs are changed to dictionary formats.

Ati-74 committed:
changing the summary stats input to dict format

summary stats inputs are changed to dict format.
calculating and finding  the neighbors...are still in D.F. format.

cheekolegend commented on Scripts/Run-SS-On-Exp&Sim/scripts/density_calculation.py:
```diff
@@ -1,53 +1,86 @@
 """
```
This density_calculation script seems to be duplicate code of the density calculation I'm writing elsewhere. The density calculation script has also been updated. Not sure how github deals with that, but maybe safer to unload this commit?

Ati-74 replied:
If I unload from this commit (assuming you mean removing it from the directory), it leads us to lose density calc. plus growth-rate-calc from the main script (due to not having the scripts in the directory to call). I think this is not what we want, right? so maybe the best is to talk in person to figure out what is efficient to do

cheekolegend replied:
Sounds good. We can merge the directories we have been using in a future PR.

cheekolegend commented on Scripts/Run-SS-On-Exp&Sim/scripts/growth_rate_stats.py:
```diff
@@ -1,39 +1,34 @@
-"""
+'''
```
This growth_rate_stat script seems to be duplicate code of the calculation I'm writing elsewhere. The growth_rate script has also been updated. Not sure how github deals with that, but maybe safer to unload this commit?

Ati-74 replied:
please see the above comment

Ati-74 committed:
CellProfiler output pickle files

Ati-74 committed:
CellModeller output pickle files

Ati-74 committed:
CellProfiler output pickle file of the last time step (K12 images)

Ati-74 committed:
fit ellipse (files' directory has been changed)

Ati-74 committed:
Estimates the fractal dimension of an image using the Minkowski–Bouligand dimension.

Ati-74 committed:
creating cell graph

this script is to make adjacency matrix, graph, and find subgraphs

Ati-74 committed:
Extracting features of bacteria

including length, orientation , vertexes, co vertexes,  minor axis length (radius), and center coordinate

Ati-74 committed:
this function is used to read experimental output pickle files

Ati-74 committed:
Calculation of dis vs growth rate

Written by A.Y

Ati-74 committed:
density calculation

Written by J.C and A.Y

Ati-74 committed:
calculation of Anisotropy

Ati-74 committed:
calculation of aspect ratio

Ati-74 committed:
calculation of fractal dimension

The first step is to draw Convex Hall around the colony of bacteria. Next, the whole Convex Hall area is filled and its image is saved, and the fractional dimension is calculated.

Ati-74 committed:
call summary state  functions and save results in dictionary

Ati-74 committed:
functionality test of fractional dimension function

This test is done on the last time step of simulation as well as experiment

Ati-74 committed:
files have been removed

Ati-74 committed:
Estimates the fractal dimension of an image using the Minkowski–Bouligand dimension.

Ati-74 committed:
calculation of fractal dimension

The first step is to draw Convex Hall around the colony of bacteria. Next, the whole Convex Hall area is filled and its image is saved, and the fractional dimension is calculated.

Ati-74 committed:
functionality test of fractional dimension function

This test is done on the last time step of simulation as well as experiment

Ati-74 committed:
Extracting features of bacteria

including length, orientation , vertexes, co vertexes,  minor axis length (radius), and center's coordinates

Ati-74 committed:
Fractal dimension summary stat is added to the unit test

Ati-74 committed:
main script to call summary stats' functions (including fractal dimension)

Ati-74 committed:
density calc plus growth-rate-stats' files are removed

Ati-74 committed:
density calculation

written by J.C and A.Y

Ati-74 committed:
calculation of dist vs growth rate

written by A.Y

cheekolegend commented on Scripts/Run-SS-On-Exp&Sim/scripts/fractal_dimension.py:
```diff
@@ -0,0 +1,68 @@
+    @param fig_name image name
+
+    """
+    # vertexes
```
# vertices

cheekolegend commented on Scripts/Run-SS-On-Exp&Sim/scripts/fractal_dimension.py:
```diff
@@ -0,0 +1,68 @@
+    return hull, points
+
+
+def calc_fractal_dimension(cells, fig_export_path, fig_name):
```
Should we calculate fractal dimension from the cell end-points or from from the colony border taken from an image? Worth discussing with @bpingalls.

bpingalls replied:
Seems the image will be a more faithful representation

Ati-74 replied:
@bpingalls Done!

cheekolegend reviewed (commented):
Very nice images. Seems like the convex hull could capture the colony border better. Should discuss how to improve the fit if we proceed with getting the colony border using the convex hull algorithm.

cheekolegend reviewed (approved):
Looks good to me. Will let @bpingalls review and merge.

Ati-74 committed:
now, the images get filled envelope of the microcolony , then fractal dimension is calculated.

Ati-74 committed:
K12 last pickle file

Ati-74 committed:
fractal dimension functionality test

Ati-74 committed:
filled image

Ati-74 committed:
input files have been removed

Ati-74 committed:
files have been removed

Ati-74 committed:
CellProfiler output pickle files

Ati-74 committed:
CellModeller output pickle files

Ati-74 committed:
main script to call the main function of each summary statistics

Ati-74 committed:
The orientation calculation method was modified

cheekolegend commented on Scripts/Run-SS-On-Exp&Sim/scripts/DataAnalysis.py:
```diff
@@ -117,17 +174,14 @@ def micro_colony_analysis(pickle_files_directory, summary_statistic_method, dt,
         # store ellipses
         ellipses = []
-        # store anisotropies of micro colonies of this time step
+        # store anisotropies of micro colonies of this rime step
```
time step

cheekolegend commented:
Fractal dimension looks good to start testing. In future developments, try to put new features in separate PRs so that different features can be merged independently. Thanks!

Ati-74 committed:
files have been removed

Ati-74 committed:
typo is fixed
